src/main/java/com/mparticle/kits/ButtonKit.java

```diff
@@ -73,148 +67,75 @@ public ButtonKit getInstance() {
-                return;
-            }
-            String referrer;
-            try {
-                referrer = URLDecoder.decode(referrerRaw, "UTF-8");
-            } catch (UnsupportedEncodingException e) {
-                // Let's fallback to raw value if decoding fails.
-                Log.e(TAG, "Failed to decode referrer '" + referrerRaw + "'", e);
-                referrer = referrerRaw;
-            }
-            ButtonLog.visibleFormat("Recorded installation with referrer. (Referrer: %s, Raw: %s).", referrer, referrerRaw);
-            mStorage.setInstallReferrer(referrer);
-        } catch (Throwable caught) {
-            ButtonLog.warn(TAG, "Exception while handling install referrer intent." + intent, caught);
+    public void setInstallReferrer(Intent intent) {
+        if (intent.getData() != null && intent.getData().getQueryParameter("btn_ref") != null) {
+            ButtonMerchant.trackIncomingIntent(getContext(), intent);
         }
     }
 
-    /**
-     * Get the Button referrer token which should be attached to any orders reported and attributed
-     * to the Button network.
-     *
-     * <pre>
-     * final ButtonKit button = (ButtonKit) MParticle.getInstance().getKitInstance(MParticle.ServiceProviders.BUTTON);
-     * if (button != null) {
-     * final String referrerToken = button.getReferrerToken();
-     * if (referrerToken != null) {
-     *      // Use token with your order reporting.
-     *  }
-     * }
-     * </pre>
-     *
-     * @return an opaque referrer string or null.
+    /*
+     * Overrides for the CommerceListener
      */
-    public String getReferrerToken() {
-        return mStorage.getReferrer();
-    }
-
-    private List<ReportingMessage> handleIntent(final Uri data, final String action) {
-        if (data == null || !Intent.ACTION_VIEW.equals(action)) {
-            return null;
-        }
 
-        handleIntentData(data);
+    @Override
+    public List<ReportingMessage> logLtvIncrease(BigDecimal bigDecimal, BigDecimal bigDecimal1,
+            String s, Map<String, String> map) {
         return null;
     }
 
-
-    protected void handleIntentData(final Uri data) {
-        /**
-         * opaque Urls (<scheme>:<scheme specific part>#<fragment>) throw on access to query parameters
-         * let's ignore all of these.
-         *
-         * @see Uri#getQueryParameter(String)
-         */
-        if (data == null || data.isOpaque()) {
-            return;
-        }
-        final String referrer = data.getQueryParameter(DeepLink.QUERY_REFERRER);
-        final String compatReferrer = data.getQueryParameter(DeepLink.QUERY_REFERRER_COMPAT);
-        final String referrerValue = referrer != null ? referrer : compatReferrer;
-        if (referrerValue == null) {
-            return;
+    @Override
+    public List<ReportingMessage> logEvent(CommerceEvent event) {
```

Comment:
Is there anything Button specific that we want to include in the ReportingMessage to mParticle?